src/app/dataset-block/metadata-block/components/event-details/event-details.component.ts

```diff
@@ -11,25 +11,26 @@ import {
     SetWatermark,
 } from "./../../../../api/kamu.graphql.interface";
 import { SupportedEvents } from "./supported.events";
-import { ChangeDetectionStrategy, ChangeDetectorRef, Component, Input, OnInit } from "@angular/core";
+import { ChangeDetectionStrategy, Component, Input } from "@angular/core";
 import { MetadataBlockFragment } from "src/app/api/kamu.graphql.interface";
-import { BaseComponent } from "src/app/common/base.component";
 import { BlockService } from "../../block.service";
 import { DatasetInfo } from "src/app/interface/navigation.interface";
+import { Observable, tap } from "rxjs";
 
 @Component({
     selector: "app-event-details",
     templateUrl: "./event-details.component.html",
     styleUrls: ["./event-details.component.sass"],
     changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class EventDetailsComponent extends BaseComponent implements OnInit {
+export class EventDetailsComponent {
     public block: MetadataBlockFragment;
```

Comment:
I suggest storing `block` field in the `BlockService` instead.
You could simply make a "currentBlock" property to access from component, as you don't need it in your template, only in methods.

Reply:
Moved `block` to `BlockService `